cmd/checkversions/checkversions.sh

```diff
@@ -0,0 +1,139 @@
+#  - parses the uscan results to determine if an update is available
+#  - tabluates the results and prints a report
+#  - returns the number of components that could be updated (fails) as the return value
+
+runtime_repo="github.com/kata-containers/runtime"
+runtime_repo_dir="$GOPATH/src/${runtime_repo}"
+versions_file="${runtime_repo_dir}/versions.yaml"
+YQ=$(which yq)
+uscan_phrase="uscan-url"
+
+# Store up the paths to all the checks that passed
+declare -a check_passed
+check_passed=()
+# Store up the paths to all the checks that failed
+declare -a check_failed
+check_failed=()
+# Associative array between path names (keys) and the current version
+declare -A current_version
+current_version=()
+declare -A newest_version
+newest_version=()
+
+# Record number of passes (up to date) and fails (could be updated)
+declare passes
+declare fails
+
+# Look in the yaml file handed to us to find uscan URLs.
+# return a list of paths to the URL items.
+#  $1 - path to versions.yaml file.
+#
+extract_uscan_items() {
+	# We convert the yaml to json, as jq can then be used to extract a list of
+	# keys that container our identifying search phrase.
+	paths=$(${YQ} r -j "$1" | jq -r 'paths | join(".")' | grep "${uscan_phrase}$")
+
+	echo "$paths"
+}
+
+main() {
+	passes=0
+	fails=0
+
+	local paths=$(extract_uscan_items "$versions_file")
+
+	for path in $paths; do
+		# Find the root key path for this component
+		local rootname="${path%.$uscan_phrase}"
+		echo "Processing [$rootname]"
+		local versionpath="$rootname.version"
+		longversion=$(yq r $versions_file $versionpath)
+		# Strip any leading non-digit values - uscan deals with digit style versions
+		# and copes better without any prefixes
+		version=$(sed -E 's/^[^0-9]+//' <<< $longversion)
+		# store away to use in the report table
+		current_version[$path]="$version"
+		uscanurl=$(yq r $versions_file $path)
+		local optspath="$rootname.uscan-opts"
+		uscanopts=$(yq r $versions_file $optspath)
+
+		# uscan needs a 'version' entry at the top of its file. uscan currently always
+		# expects that to be version 4.
+		echo "version=4" > watchfile
+		if [ -n "$uscanopts" ] && [ "$uscanopts" != "null" ]; then
+			echo "$uscanopts  \\" >> watchfile
+		fi
+		echo "$uscanurl" >> watchfile
+
+		# And run the uscan
+		result="$(uscan --report --package ${rootname} --upstream-version ${version} --watchfile watchfile 2>/dev/null)"
+		
+		# Extract the latest version found
+		newversion="$(egrep -o 'remote site is (.*)+,' <<< $result | sed 's/remote site is //' | sed 's/,//')"
+		newest_version[$path]="$newversion"
+
+		# And note if that was marked as being 'Newer' than our current version
+		failure=$(grep Newer <<< $result)
+
+		# Store the results for later
+		if [ -n "$failure" ]; then
+			((fails++))
+			check_failed[${#check_failed[@]}]="$path"
+		else
+			((passes++))
+			check_passed[${#check_passed[@]}]="$path"
+		fi
+	done
+}
+
+# Generate a nice tabluated summary of the results.
```

Comment:
s/tabluated/tabulated/ ?

Reply:
heh, and I think I did it twice... fixing. will wait for any other fold ins...

Reply:
typos fixed and pushed..